connect/src/spirc.rs

```diff
@@ -270,54 +272,141 @@ impl Spirc {
         Ok((spirc, task.run()))
     }
 
+    /// Safely shutdowns the spirc.
+    ///
+    /// This pauses the playback, disconnects the connect device and
+    /// bring the future initially returned to an end.
+    pub fn shutdown(&self) -> Result<(), Error> {
+        Ok(self.commands.send(SpircCommand::Shutdown)?)
```

Comment:
I'm not bent on it, but think I would have re-used the `Result` instead of wrapping it in an `Ok`.
Like: `self.commands.send(Foo::Bar)` optionally with `.map_err(Into::into)`.

Reply:
> I didn't touch the code itself but moved it a bit around

The problem you will find is, that the inner method call returns the incorrect error and rust doesn't implicitly converts the returned type. You have to convert the error by calling `.map_err(Into::into)`, or using `?` like it is done here.

As I only moved the code I will not change it. But it's a bit tricky which way you should use in those cases as both have the same result but just look different.